src/rhsm/connection.py

```diff
@@ -500,10 +500,14 @@ class RateLimitExceededException(RestlibException):
     The retry_after attribute may not be included in the response.
     """
 
-    def __init__(self, code: int, msg: str = None, headers: str = None) -> None:
+    def __init__(self, code: int, msg: str = None, headers: dict = None) -> None:
```

Comment:
The type hint of `headers` was just wrong, right? I can see that this exception is raised in `validateResult()` and it seems that the `headers` is dict there.